### PR TITLE
Rename Python package "neovim" to "pynvim"

### PR DESCRIPTION
**Pull request by mhinz**
For a moment I pondered the idea of adding lots of glue code so that both, `neovim` and `pynvim`, get considered (since the former acts as a transition package for the latter), but I do not think it's worth the effort.

(And since it's directly related to Neovim, maybe the rename should be mentioned at https://github.com/neovim/neovim/wiki/Following-HEAD as well.)

### CODE REVIEW

File: runtime/autoload/health/provider.vim [hunk 1]
```diff
@@ -352,7 +352,7 @@ function! s:check_python(version) abort
       call health#report_warn('pyenv is not set up optimally.', [
             \ printf('Create a virtualenv specifically '
             \ . 'for Neovim using pyenv, and set `g:%s`.  This will avoid '
-            \ . 'the need to install the Neovim Python module in each '
+            \ . 'the need to install the pynvim Python module in each '
```
Comment (jamessan) [inline runtime/autoload/health/provider.vim]: The previous wording was referring to Neovim's python module, so it should either be maintained or changed to `pynvim module`.  Similarly on line 369.

**Comment by jamessan** [comment]
> And since it's directly related to Neovim, maybe the rename should be mentioned at [neovim/neovim/wiki/Following-HEAD](https://github.com/neovim/neovim/wiki/Following-HEAD) as well

Agreed, especially if this gets merged.

**Review by justinmk — APPROVED** [review]

**Comment by justinmk** [comment]
Is there a simple check (and resolution) that could be added to `:checkhealth`?

**Comment by mhinz** [comment]
> Is there a simple check (and resolution) that could be added to `:checkhealth`, if `neovim` module is still present?

I'll open a new PR for that.

**Comment by ibash** [comment]
👋 I just upgraded to neovim 0.3.2 with homebrew and would get errors on startup until I installed `pynvim`. It might be worth making a message more prevalent.

**Comment by justinmk** [comment]
For others finding this issue, see also [this note](https://github.com/neovim/neovim/wiki/Following-HEAD#20181118).

**Pull request merged**